packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -0,0 +1,380 @@
+
+  /// The color to use for the check icon when this checkbox is checked.
+  ///
+  /// If null, then the value of [CupertinoColors.white] is used.
+  final Color? checkColor;
+
+  /// If true the checkbox's [value] can be true, false, or null.
+  ///
+  /// [CupertinoCheckbox] displays a dash when its value is null.
+  ///
+  /// When a tri-state checkbox ([tristate] is true) is tapped, its [onChanged]
+  /// callback will be applied to true if the current value is false, to null if
+  /// value is true, and to false if value is null (i.e. it cycles through false
+  /// => true => null => false when tapped).
+  ///
+  /// If tristate is false (the default), [value] must not be null.
+  final bool tristate;
+
+  /// The color for the checkbox's border shadow when it has the input focus.
+  ///
+  /// If null, then a paler form of the [activeColor] will be used.
+  final Color? focusColor;
+
+  /// {@macro flutter.widgets.Focus.focusNode}
+  final FocusNode? focusNode;
+
+  /// {@macro flutter.widgets.Focus.autofocus}
+  final bool autofocus;
+
+  /// The color and width of the checkbox's border.
+  ///
+  /// If this property is null, then the side will be width 1.
+  final BorderSide? side;
+
+  /// The shape of the checkbox.
+  ///
+  /// If this property is null then the shape will be a [RoundedRectangleBorder]
+  /// with a circular corner radius of 4.0.
+  final OutlinedBorder? shape;
+
+  /// The width of a checkbox widget.
+  static const double width = 18.0;
+
+  @override
+  State<CupertinoCheckbox> createState() => _CupertinoCheckboxState();
+}
+
+class _CupertinoCheckboxState extends State<CupertinoCheckbox> with TickerProviderStateMixin, ToggleableStateMixin {
+  final _CheckboxPainter _painter = _CheckboxPainter();
+  bool? _previousValue;
+
+  bool focused = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _previousValue = widget.value;
+  }
+
+  @override
+  void didUpdateWidget(CupertinoCheckbox oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.value != widget.value) {
+      _previousValue = oldWidget.value;
+    }
+  }
+
+  @override
+  void dispose() {
+    _painter.dispose();
+    super.dispose();
+  }
+
+  @override
+  ValueChanged<bool?>? get onChanged => widget.onChanged;
+
+  @override
+  bool get tristate => widget.tristate;
+
+  @override
+  bool? get value => widget.value;
+
+  void onFocusChange(bool value) {
+    if (focused != value) {
+      focused = value;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+
+    final Color effectiveActiveColor = widget.activeColor
+      ?? CupertinoColors.activeBlue;
+    final Color? inactiveColor = widget.inactiveColor;
+    final Color effectiveInactiveColor = inactiveColor
+      ?? CupertinoColors.inactiveGray;
+
+    final Color effectiveFocusOverlayColor = widget.focusColor
+      ?? HSLColor
+          .fromColor(effectiveActiveColor.withOpacity(0.80))
+          .withLightness(0.69).withSaturation(0.835)
```

Comment:
It would be nice if these magic numbers were private const variables at the top of the file (starting with `_k`). Names give more context about the value than the value does.

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-mysterious-and-magical-numbers-that-lack-a-clear-derivation